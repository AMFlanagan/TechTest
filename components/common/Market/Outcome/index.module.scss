.outcome {
  // reset button styles
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;

  // styles
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  background-color: #222222;
  text-align: center;
  width: 100%;
  padding: 16px;
}

.selected,
.outcome:hover {
  background-color: green;
}

.odds,
.name {
  padding: 4px 0;
  align-self: center;
}
