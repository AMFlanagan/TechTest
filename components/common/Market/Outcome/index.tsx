import React from 'react'

import { useBetSlip } from '../../../../context/BetSlip'

import styles from './index.module.scss'

interface IProps {
  outcome: string
  marketId: number
  onSelect: (outcome: string, inBetSlip: boolean) => void
}

const Outcome: React.FC<IProps> = (props: IProps) => {
  const { state } = useBetSlip()

  const isInBetSlip = () =>
    !!state.bets.find(
      (bet: any) =>
        bet.outcome === props.outcome && bet.marketId === props.marketId
    )

  return (
    <button
      onClick={() => props.onSelect(props.outcome, isInBetSlip())}
      className={`${styles.outcome} ${isInBetSlip() && styles.selected}`}
    >
      <div className={styles.name}>{props.outcome}</div>
    </button>
  )
}

Outcome.displayName = 'Outcome'

export default Outcome
