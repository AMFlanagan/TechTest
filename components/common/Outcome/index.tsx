import React from 'react'
import { IOutcome } from '../../../types'
import formatOdds from '../../../utils/formatOdds'

import styles from './index.module.scss'

interface IProps {
  outcome: IOutcome
}

const Outcome: React.FC<IProps> = (props: IProps) => {
  return (
    <button className={styles.outcome}>
      <div className={styles.name}>{props.outcome.name}</div>
      <div className={styles.odds}>{formatOdds(props.outcome.odds)}</div>
    </button>
  )
}

Outcome.displayName = 'Outcome'

export default Outcome
