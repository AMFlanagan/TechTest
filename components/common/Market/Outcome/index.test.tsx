import Outcome from '.'

import { fireEvent, render } from '@testing-library/react'

const mockHook = {
  state: {
    bets: [
      {
        marketId: 2,
        outcome: 'Team B Win'
      }
    ]
  },
}

jest.mock('../../../../context/BetSlip', () => ({
  useBetSlip() {
    return mockHook
  },
}))

describe('Outcome', () => {
  it('renders as expect', () => {
    const outcome = 'Team A Win'

    const { asFragment, getByText } = render(
      <Outcome onSelect={() => {}} marketId={1} outcome={outcome} />
    )

    expect(getByText('Team A Win')).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })

  it('calls onSelect when clicked and not selected', () => {
    const outcome = 'Team A Win'
    const onSelectMock = jest.fn()

    const { getByText } = render(
      <Outcome onSelect={onSelectMock} marketId={1} outcome={outcome} />
    )

    fireEvent.click(getByText('Team A Win'))

    expect(onSelectMock).toBeCalledWith(
      'Team A Win',
      false
    )
  })

  it('calls onSelect when clicked and selected', () => {
    const outcome = 'Team B Win'
    const onSelectMock = jest.fn()

    const { getByText } = render(
      <Outcome onSelect={onSelectMock} marketId={2} outcome={outcome} />
    )

    fireEvent.click(getByText('Team B Win'))

    expect(onSelectMock).toBeCalledWith(
      'Team B Win',
      true
    )
  })
})
