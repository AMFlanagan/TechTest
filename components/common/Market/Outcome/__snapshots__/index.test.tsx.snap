// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outcome renders as expect 1`] = `
<DocumentFragment>
  <button
    class="outcome false"
  >
    <div
      class="name"
    >
      Team A Win
    </div>
  </button>
</DocumentFragment>
`;
